src/components/clusterDetail/ClusterInstallationProgressCard.tsx

```diff
@@ -0,0 +1,126 @@
+
+  const fetchCredentials = React.useCallback(() => {
+    const fetch = async () => {
+      setCredentialsError(undefined);
+      try {
+        const response = await getClusterCredentials(cluster.id);
+        setCredentials(response.data);
+      } catch (err) {
+        setCredentialsError(err);
+      }
+    };
+    fetch();
+  }, [cluster.id]);
+
+  React.useEffect(() => {
+    if (cluster.status === 'installed') {
+      fetchCredentials();
+    }
+  }, [cluster.status, fetchCredentials]);
+
+  return (
+    <>
+      <AlertsContextProvider>
+        <CancelInstallationModal
+          isOpen={cancelInstallationModalOpen}
+          onClose={() => setCancelInstallationModalOpen(false)}
+          clusterId={cluster.id}
+        />
+        <ResetClusterModal
+          isOpen={resetClusterModalOpen}
+          onClose={() => setResetClusterModalOpen(false)}
+          cluster={cluster}
+        />
+      </AlertsContextProvider>
+      <Card id="ai-cluster-details-card" isExpanded={isCardExpanded}>
+        <CardHeader
+          onExpand={() => setIsCardExpanded(!isCardExpanded)}
+          toggleButtonProps={{
+            id: 'toggle-ai-cluster-details',
+            'aria-label': 'AI Cluster Details',
+            'aria-labelledby': 'titleId toggle-button',
+            'aria-expanded': isCardExpanded,
+          }}
+        >
+          <Title headingLevel="h1" size="lg" className="card-title">
+            Installation progress
+          </Title>
+          {!isCardExpanded && (
+            <div className="pf-u-ml-md">
+              <ClusterStatusIcon status={cluster.status} />
+            </div>
+          )}
+        </CardHeader>
+        <CardBody>
+          <ClusterProgress cluster={cluster} minimazedView={!isCardExpanded} />
+          <ClusterDetailsButtonGroup cluster={cluster} />
+        </CardBody>
+        <CardExpandableContent>
+          <CardBody>
+            {['installed', 'installing', 'finalizing'].includes(cluster.status) && (
+              <FailedHostsWarning cluster={cluster} />
+            )}
+            {cluster.status === 'error' && (
+              <ClusterInstallationError
+                cluster={cluster}
+                setResetClusterModalOpen={setResetClusterModalOpen}
+              />
+            )}
+            {cluster.status === 'cancelled' && (
+              <ClusterInstallationError
+                title="Cluster installation was cancelled"
+                cluster={cluster}
+                setResetClusterModalOpen={setResetClusterModalOpen}
+              />
+            )}
+            {cluster.status === 'installed' && (
+              <ClusterCredentials
+                cluster={cluster}
+                credentials={credentials}
+                error={!!credentialsError}
+                retry={fetchCredentials}
+                idPrefix={getID('cluster-creds')}
+              />
+            )}
+            <div className="vertical-margin">
```

Comment:
Do we need that extra div? Would it make sense to pass the class to the `HostsTable`?